scalyr_agent/monitor_utils/k8s.py

```diff
@@ -2160,21 +2165,33 @@ def __init__(
         self._session.headers.update(headers)
 
         global_log.info("KubeletApi host ip = %s" % self._host_ip)
-        self._kubelet_url = self._build_kubelet_url(kubelet_url_template, host_ip)
+        self._kubelet_url = self._build_kubelet_url(
+            kubelet_url_template, host_ip, node_name
+        )
         self._fallback_kubelet_url = self._build_kubelet_url(
-            FALLBACK_KUBELET_URL_TEMPLATE, host_ip
+            FALLBACK_KUBELET_URL_TEMPLATE, host_ip, node_name
         )
         self._timeout = 20.0
 
     @staticmethod
-    def _build_kubelet_url(kubelet_url, host_ip):
-        if host_ip:
-            return kubelet_url.substitute(host_ip=host_ip)
+    def _build_kubelet_url(kubelet_url, host_ip, node_name):
+        if node_name and host_ip:
+            return kubelet_url.substitute(node_name=node_name, host_ip=host_ip)
         return None
 
     def _switch_to_fallback(self):
         self._kubelet_url = self._fallback_kubelet_url
 
+    def _verify_connection(self):
+        """ Return whether or not to use SSL verification
+        """
+        if self._verify_https and self._ca_file:
```

Comment:
Should we throw inside the constructor where we set the config values if ``verify_https`` is true, but ``ca_file`` doesn't exist (or is None)?

Or will we just outsource that to requests?